ihmc-perception/src/main/java/us/ihmc/perception/detections/yolo/YOLOv8DetectionExecutor.java

```diff
@@ -137,28 +137,29 @@ public void runYOLODetectionOnAllModels(RawImage colorImage, RawImage depthImage
 
    /**
     * Non-blocking call to run YOLO on the provided images
-    * @param colorImage BGR color image, used for YOLO detection
+    * @param colorImage RGB color image, used for YOLO detection
     * @param depthImage 16UC1 depth image, used to get points of detected objects
     */
    public void runYOLODetection(YOLOv8ObjectDetector yoloDetector, RawImage colorImage, RawImage depthImage)
    {
       if (yoloDetector.isReady() && !yoloExecutorService.isShutdown())
       {
+         // Acquire the images
+         if (colorImage.get() == null || depthImage.get() == null)
+            return;
+
          yoloExecutorService.submit(() ->
          {
-            // Acquire the images
-            if (!colorImage.isAvailable() || !depthImage.isAvailable())
-               return;
-            colorImage.get();
-            depthImage.get();
-
             // Run YOLO to get results
             YOLOv8DetectionResults yoloResults = yoloDetector.runOnImage(colorImage, yoloConfidenceThreshold, yoloNMSThreshold, yoloMaskThreshold);
 
             // TODO: temp hack
-            if (yoloDetectionResults.containsKey(lastRunDetectorIndex))
-               yoloDetectionResults.remove(lastRunDetectorIndex).destroy();
-            yoloDetectionResults.put(lastRunDetectorIndex, yoloResults);
+            synchronized (yoloDetectionResults)
```

Comment:
I think the detection results may have been getting deallocated while the annotated image thread was using them, so I put think synchronized block for now. 

Really, this class needs an overhaul.